src/usage.rs

```diff
@@ -0,0 +1,23 @@
+//! The module for the usage advice.
+//!
+//! The usage advice defined in this module is specialised regarding the needs
+//! of this application.  This is,
+//!
+//! * the application name is hard coded,
+//! * the tool tip is hard coded, and
+//! * the expected order of arguments in the usage template is hard coded.
+
+use getopts::Options;
+
+/// A small in-app documentation.
+///
+/// * `options` - The options the render the tool tips for.
+///
+/// This function will write a brief usage information, including a short
+/// introduction to the meaning of the configured options, to `stdout`.
+pub fn usage(options: &Options) {
+    println!(
+        "Usage:  commit-analyzer <FILE> [OPTIONS]\n{}",
```

Comment:
I began the set of changes with the fix of a typo:  `commit-analzer`.  One had forgotten the letter `y`.  In order to prevent similar typos in the future, I looked for an appropriate solution.

Making the name of the application a constant such that you only need to care for the correctness of your constant would help.  But then, you could also hard code the name directly into the corresponding function since it cannot be reused in another project due to having a path to a file as a mandatory argument.

Making it completely flexible and, hence, reusable would have meant a parameter overkill.  You have needed to pass

* the application's name,
* its tool tip,
* its options, and
* the usage line.

Hard coding the app name seemed to be most effective.

Reply:
I removed one `println` because it is a function call which can be omitted by just combining the two texts.  That way, a slight performance boost is achieved.  (Obviously, it is cheaper in terms of performance to call only one function instead of two.)

I decided to still use `println` since `cargo` uses the same formatting for its option documentation; see `cargo --help` for details or, more precisely, the blank line before between its output and your prompt.

Reply:
Yeah, lke I said, I'm fine with this change.

> Also you removed a line feed, since `println` will add a `\n` at the end of your format string.

Please fix that. While this might seem petty, I think pull request should be similarly atomic like commits. So accidental changes should be reverted here.

Reply:
> I removed one `println` because it is a function call which can be omitted by just combining the two texts. That way, a slight performance boost is achieved. (Obviously, it is cheaper in terms of performance to call only one function instead of two.)

Few things. Firstly, println is not a function, but a macro.
Secondly, I actually checked the assembler output, because I thought the llvm would be able to optimize this into the same code. But I think it fails to do that, since println also flushes its writer and this may lead to difference to other programs reading from its output. I think. But in the end the difference is a single `call` instruction, which is not perceivable in any way, since any natural fluctuation is worse then that. Also this is a **very** cold code path, so that performance boosts here are not important at all.

But I'm fine with keeping it as one `println!`. But performance would not be an argument in favor of that. 😅

Reply:
I just fixed it.